spec/models/manageiq/providers/redhat/infra_manager/provision/configuration/network_spec.rb

```diff
@@ -321,10 +330,29 @@
             expect(@task.get_mac_address_of_nic_on_requested_vlan).to eq(nil)
           end
         end
+
+        context "'profile_id (network_name)' is specified" do
+          before do
+            assign_vnic_profile(get_profile_description(vnic_profile_name, network.name))
+          end
+
+          it 'nics list contains a nic with the specified profile description' do
+            expect(@task.get_mac_address_of_nic_on_requested_vlan).to eq(mac_address)
+          end
+
+          it 'nics list does not contain a nic with the specified profile description' do
```

Comment:
If I understand the test correctly, the "it" statement is actually a context - "it" should specify what is the expected result, while context describe in what context it happens. (same for previous "it")